lib/openhab/dsl/types/datetime.rb

```diff
@@ -0,0 +1,339 @@
+        end
+
+        #
+        # Returns true if time represents a time in UTC (GMT)
+        #
+        # @return [Boolean] true if utc_offset == 0, false otherwise
+        #
+        def utc?
+          utc_offset.zero?
+        end
+
+        #
+        # The timezone
+        #
+        # @return [String] The timezone in `[+-]hh:mm(:ss)` format ('Z' for UTC) or nil if the Item has no state
+        #
+        def zone
+          zoned_date_time.zone.id
+        end
+
+        #
+        # Check if missing method can be delegated to other contained objects
+        #
+        # @param [String, Symbol] meth the method name to check for
+        #
+        # @return [Boolean] true if DateTimeType, ZonedDateTime or Time responds to the method, false otherwise
+        #
+        def respond_to_missing?(meth, *)
+          @datetime.respond_to?(meth) ||
+            zoned_date_time.respond_to?(meth) ||
+            Time.instance_methods.include?(meth.to_sym)
+        end
+
+        #
+        # Forward missing methods to the OpenHAB DateTimeType, its ZonedDateTime object or a ruby Time
+        # object representing the same instant
+        #
+        # @param [String] meth method name
+        # @param [Array] args arguments for method
+        # @param [Proc] block <description>
+        #
+        # @return [Object] Value from delegated method in OpenHAB NumberItem
+        #
+        def method_missing(meth, *args, &block)
+          if @datetime.respond_to?(meth)
+            @datetime.__send__(meth, *args, &block)
+          elsif zoned_date_time.respond_to?(meth)
+            zoned_date_time.__send__(meth, *args, &block)
+          elsif Time.instance_methods.include?(meth.to_sym)
+            to_time.send(meth, *args, &block)
+          else
+            raise NoMethodError, "undefined method `#{meth}' for #{self.class}"
+          end
+        end
+
+        private
+
+        #
+        # Converts other objects to a DateTimeType
+        #
+        # @param [String, Numeric, Time] datetime an object that can be parsed or converted into
+        #   a DateTimeType
+        #
+        # @return [Java::org::openhab::core::library::types::DateTimeType] Object representing the same time
+        #
+        def datetime_type_from(datetime)
+          case datetime
+          when String
+            parse(datetime)
+          when Numeric
+            datetime_type_from_numeric(datetime)
+          when Time
+            datetime_type_from_time(datetime)
+          else
+            raise "Unexpected type #{datetime.class} provided to DateTime initializer"
+          end
+        end
+
+        #
+        # Converts a Numeric into a DateTimeType
+        #
+        # @param [Numeric] numeric A Integer or Float representing the number of seconds since the epoch
+        #
+        # @return [Java::org::openhab::core::library::types::DateTimeType] Object representing the same time
+        #
+        def datetime_type_from_numeric(numeric)
```

Comment:
is this coerce?  (asking, not sure).

Reply:
This is used by the initialize method to create a new DateTimeType from a numeric passed to initialize. See below for further discussion.

Reply:
Ok..  I thought they had external usage.

Should they all be private then?

Reply:
I have changed them to static methods, and it feels better actually :)